ql/src/java/org/apache/hadoop/hive/ql/plan/CopyWork.java

```diff
@@ -136,4 +140,18 @@ public boolean isOverwrite() {
   public void setOverwrite(boolean overwrite) {
     this.overwrite = overwrite;
   }
+
+  public void setLoadTableStateWrapper(LoadTableStateWrapper loadTableStateWrapper) {
+    this.loadTableStateWrapper = loadTableStateWrapper;
+  }
+
+  public void setValuesBeforeExec() throws HiveException {
+    if (loadTableStateWrapper == null) {
+      return;
+    }
+
+    Table table = loadTableStateWrapper.getTableIfExists();
```

Comment:
CopyWork was generic in nature. setValuesBeforeExec is tying it to resolve the path only from table. All we need is to be able to resolve the path dynamically. A resolver interface should be prefered.